R/plotPValHist.R

```diff
@@ -16,63 +17,78 @@
 #' @examples
 #' \dontrun{
 #'    # Print to console using all defaults
-#'    MyPvalMatrix <- extractCol(getType(myDGEobj, "topTable"), "P.Value")
-#'    plotPvalHist(MyPvalMatrix)
+#'    plotPvalHist(DGEdata, P.Val)
 #'
 #'    # Use some custom arguments
-#'    myplot <- plotPvalHist(MyPValMatrix)
+#'    myplot <- plotPvalHist(DGEdata, P.Val)
 #' }
 #'
 #' @import ggplot2
 #' @importFrom dplyr filter select
 #' @importFrom canvasXpress canvasXpress
 #'
 #' @export
-plotPvalHist <- function(P.Val,
-                         plotType = "canvasXpress",
+plotPvalHist <- function(DGEdata,
+                         P.Val          = "P.Value",
+                         plotType       = "canvasXpress",
                          facet          = TRUE,
                          binWidth       = 0.02,
                          transparency   = 0.6,
                          color    = "dodgerblue3") {
 
-    plotType = tolower(plotType)
+    assertthat::assert_that(!missing(DGEdata),
+                            !is.null(DGEdata),
+                            "DGEobj" %in% class(DGEdata),
+                            msg = "DGEdata must be specified and must belong to DGEobj class.")
 
-    assertthat::assert_that(!missing(P.Val),
-                            class(P.Val)[[1]] %in% c("matrix","data.frame"),
-                            all(sapply(data.frame(P.Val),is.numeric)),
-                            msg = "P.Val must be specified and must be of class matrix or dataframe and must contain only numeric values.")
+    if (any(is.null(P.Val),
+            !is.character(P.Val),
+            length(P.Val) != 1)) {
+        warning("P.Val must be a singular value of class character. Assigning default value 'P.Value'.")
+        P.Val <- "P.Value"
+    }
 
-    assertthat::assert_that(all(sapply(data.frame(P.Val),is.numeric)),
-                            msg = "P.Val must contain only numeric values.")
+    P.Val <- extractCol(getType(DGEdata, "topTable"), colName = P.Val, robust = FALSE)
```

Comment:
need to mention what packages function `getType` and `extractCol`  belongs to

Reply:
done